mitiq/lre/tests/test_multivariate_richardson.py

```diff
@@ -0,0 +1,221 @@
+# Copyright (C) Unitary Fund
+#
+# This source code is licensed under the GPL license (v3) found in the
+# LICENSE file in the root directory of this source tree.
+
+"""Unit tests for multivariate extrapolation inference functions."""
+
+import numpy as np
+import pytest
+from cirq import Circuit, LineQubit, ops
+
+from mitiq.lre.inference.multivariate_richardson import (
+    _full_monomial_basis_term_exponents,
+    linear_combination_coefficients,
+    sample_matrix,
+)
+from mitiq.lre.multivariate_scaling.layerwise_folding import (
+    multivariate_layer_scaling,
+)
+
+qreg1 = LineQubit.range(3)
+test_circuit1 = Circuit(
+    [ops.H.on_each(*qreg1)],
+    [ops.CNOT.on(qreg1[0], qreg1[1])],
+)
+test_circuit2 = Circuit(
+    [ops.H.on_each(*qreg1)],
+    [ops.CNOT.on(qreg1[0], qreg1[1])],
+    [ops.X.on(qreg1[2])],
+    [ops.TOFFOLI.on(*qreg1)],
+)
+
+
+@pytest.mark.parametrize(
+    "test_num_layers, test_degree, expected",
+    [
+        (1, 1, [(0,), (1,)]),
+        (
+            2,
+            2,
+            [(0, 0), (1, 0), (0, 1), (2, 0), (1, 1), (0, 2)],
+        ),
+        (
+            3,
+            2,
+            [
+                (0, 0, 0),
+                (1, 0, 0),
+                (0, 1, 0),
+                (0, 0, 1),
+                (2, 0, 0),
+                (1, 1, 0),
+                (0, 2, 0),
+                (1, 0, 1),
+                (0, 1, 1),
+                (0, 0, 2),
+            ],
+        ),
+    ],
+)
+def test_basis_exp(test_num_layers, test_degree, expected):
+    assert (
+        _full_monomial_basis_term_exponents(test_num_layers, test_degree)
+        == expected
+    )
+
+
+@pytest.mark.parametrize(
+    "test_num_layers, test_degree",
+    [(1, 1), (2, 2), (3, 2), (10, 4)],
+    # Note need to add (100, 2) here. This makes the unit test very slow.
```

Comment:
This should either be removed, or we'll need to refactor `_full_monomial_basis_term_exponents` to speed it up. If we need to speed it up, what are the largest values it should be performant at?

Reply:
> we'll need to refactor _full_monomial_basis_term_exponents to speed it up

I was thinking of leaving this for the immediate future.

Reply:
Sounds good. Let's finish the full implementation and then come back to it. Please add `TODO` to the comment so it's a bit more discoverable.